Add information regarding key rotation during backups for pg_basebackup making servers fail to start

Andriciuc opened the pull request:
- add as known issue in release notes
- fix a broken link in features.md (not related to issue...)
- add to global key providers a warning about keyring provider with WAL encrypt
- add new subtopic in Backup WAL about key rotations during backups for  file-based key providers

Based on PG-1895 description.

Andriciuc committed:
add information regarding pg_basebackup making servers fail to start during wal replay

- add as known issue in release notes
- fix a broken link in features.md
- add to global key providers a warning about keyring provider with WAL encrypt
- add new subtopic in Backup WAL about key rotations during backups for  file-based key providers

dAdAbird commented on contrib/pg_tde/documentation/docs/global-key-provider-configuration/overview.md:
```diff
@@ -11,6 +11,11 @@ To use an external KMS with `pg_tde`, follow these two steps:
      While key files may be acceptable for **local** or **testing environments**, KMS integration is the recommended approach for production deployments.
 
+!!! warning
+    When using the file keyring provider with WAL encryption, avoid creating new providers that reference different keyring files while `pg_basebackup` is running.  
```
It is related to all types of provides. For example, creating a new Vault with different `vault_url` or `vault_mount_path` would result in the same issue.

dAdAbird replied:
And just creating new providers doesn't do any harm. Errors may happen on when keys rotated with such providers. 
But I would not go into such depth, just "don't rotate keys during backups" will do as for me.

Andriciuc replied:
Ooh! Got it! Updated to keep it simple:

Do not rotate encryption keys while `pg_basebackup` is running. Standbys or standalone clusters created from such backups may fail to start during WAL replay. Schedule rotations outside your backup windows and run a new full backup afterward.

dAdAbird commented on contrib/pg_tde/documentation/docs/global-key-provider-configuration/overview.md:
```diff
@@ -11,6 +11,11 @@ To use an external KMS with `pg_tde`, follow these two steps:
+!!! warning
+    When using the file keyring provider with WAL encryption, avoid creating new providers that reference different keyring files while `pg_basebackup` is running.  
+
+    Standbys initialized from such backups may fail to start during WAL replay. Schedule rotations outside your backup windows and run a new full backup afterward.
```
Not only standbys, starting standalone from such a backup will fail as well.

Andriciuc replied:
Added standalone clusters too

dAdAbird commented on contrib/pg_tde/documentation/docs/how-to/backup-wal-enabled.md:
````diff
@@ -21,6 +21,22 @@ Also copy any external files referenced by your providers configuration (such as
+```
+
+To ensure standby recoverability, plan key rotations outside backup windows or take a new full backup after rotation completes.
+
````
The same comments as for the section above

Andriciuc replied:
Updated! Now it has tighter scope, less info thrown in your face (I wanted to remove the specifics so we keep it simple) and included standalones in the warning.

dAdAbird commented on contrib/pg_tde/documentation/docs/release-notes/release-notes-v2.0.md:
```diff
@@ -30,14 +30,18 @@ Clusters that did not use WAL encryption in beta can be upgraded normally.
 
 ## Known issues
 
+* When creating new file-based key providers that reference different keyring files while `pg_basebackup` is running can result in a standby that fails to start (key retrieval errors during WAL replay).
+
+    To avoid this, perform key rotations before or after backups, and take a new full backup once rotation completes.
+
```
The same comments as for the section above

Andriciuc replied:
changed to:

* Rotating encryption keys while `pg_basebackup` is running may cause standbys or standalone clusters initialized from the backup to fail during WAL replay.

    Avoid key rotations during backups. Run a new full backup after completing a rotation.

Andriciuc committed:
update the warnings to be simpler and remove file provider as it pertains to ALL providers

- Added new known issue in RN about basebackup using fetch will lead to errors
- updated sql error to say provider instead of local to be more generic
- removed excessive explanations and kept the subtopic simple and to the point

Andriciuc committed:
Merge branch 'release-17.5.3' into doc-basebackup-server-fail

dAdAbird commented on contrib/pg_tde/documentation/docs/how-to/backup-wal-enabled.md:
```diff
@@ -21,6 +21,22 @@ Also copy any external files referenced by your providers configuration (such as
+!!! warning
+    Do not rotate SMGR or WAL encryption keys while `pg_basebackup` is running. Standbys or standalone clusters created from such backups may fail to start during WAL replay.
+
+Rotations during a base backup can leave the standby in an inconsistent state where it cannot retrieve the correct key history.
```
Sounds a bit ambiguous to me. I can read it as if the server from which backup is taken can fail. As in fact all rotations can break is a backup.

Andriciuc replied:
Hmm... I went too general this time, how about this 🤔?

Do not rotate SMGR or WAL encryption keys while `pg_basebackup` is running.  Backups taken during a key rotation may be unusable: any standby or standalone cluster initialized from such a backup can fail to start during WAL replay.

Rotating keys mid-backup can produce an inconsistent backup that does not contain the correct key history.

Andriciuc committed:
Merge branch 'release-17.5.3' into doc-basebackup-server-fail
